### ISSUE TITLE
Crash when running jekyll with --incremental

### ISSUE THREAD

**Issue by jonsterling** [paolobrasolin/jekyll-antex#19]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behaviour

I am running `bundle exec jekyll serve --incremental`. I receive a crash here:

```
/Users/jon/.gem/ruby/2.7.5/gems/jekyll-antex-0.6.0/lib/jekyll/antex.rb:24:in `inject_style_attributes': undefined method `gsub' for nil:NilClass (NoMethodError)
```

This does not happen if I don't run with `--incremental`.

### Expected behaviour

It should not crash.

### Steps to reproduce

This can be observed by cloning https://github.com/jonsterling/math and running `bundle exec jekyll serve --incremental`.

I have "fixed" it by changing [https://github.com/paolobrasolin/jekyll-antex/blob/main/lib/jekyll/antex.rb#L30](this line) in my local installation to say `(output || "")`, but I don't know if this is the desired behavior because I don't know the context of the code. Anyway, let me know what you think :)

**Comment by jonsterling** [comment]
Just to update, it looks like my solution breaks antex from compiling tex blocks.

**Comment by paolobrasolin** [comment]
Thanks for the report @jonsterling!

Well, there are two interacting problems here:
1. `jekyll-antex` never supported `--incremental`, and
2. your extension [`Jekyll::ImportTag`](https://github.com/jonsterling/math/blob/c4d813e29f999c9fd25ef5f549818a102080bfd2/_plugins/import.rb#L71-L99) doesn't support `--incremental` either.

First of all, `--incremental` is an experimental and limited feature -- see the [documentation](https://jekyllrb.com/docs/configuration/incremental-regeneration/).
As you can imagine it builds a graph to track dependencies, but it does so **only** for layouts and `include` tags.


One is tempted to replace your `import` tags with ordinary `include` tags. The problem is that they [load files directly from disk and stores them as cached "partials"](https://github.com/jekyll/jekyll/blob/1aea1580c193c271b8cc083607105da363af358f/lib/jekyll/tags/include.rb#L140-L158). This means their content doesn't go through the usual Jekyll pipeline. Instead, it just gets interpolated verbatim during the rendering phase. Obviously that happens after the `pre_render` hook, which is used by `jekyll-antex` to replace "aliases" with its primitive `antex` liquid tag.

The result is that `jekyll-antex` never gets the chance to replace its "aliases" and all its content within the `include`d files is ignored. On the other hand the `antex` tag itself works well; so, as long as you don't use aliases (which is terrible) `jekyll-antex` works fine. That is, except for the small bug you already found and fixed.

I never bothered fixing all this because the `include` feature feels like it's just to handle fragments of layout, not page content. (E.g. how should the metadata of the `include`d files even be treated for this to be generally well behaved?)

I can think of some workarounds, but the best I can come up with is monkey-patching `Jekyll::Tags::IncludeTag` [here](https://github.com/jekyll/jekyll/blob/1aea1580c193c271b8cc083607105da363af358f/lib/jekyll/tags/include.rb#L174-L177), which kinda sucks.

That's why I'm not sure even trying to support `--incremental` is even a good idea, and that explains point 1.

As for point 2: your `import` tag isn't doing the bookkeeping that `include` does, which is what's enabling the `--incremental` behaviour in the first place. Specifically, I'm talking about `add_include_to_dependency` in https://github.com/jekyll/jekyll/blob/1aea1580c193c271b8cc083607105da363af358f/lib/jekyll/tags/include.rb

So, to recap:
* yes, the bug you found is real and I just published a fix so at least it doesn't crash;
* I'm not sure how to make `jekyll-antex` behave nicely with `--incremental` behaviour because I think Jekyll lacks the internal infrastructure to make it ~easy~ not a terrible kludge;
* perhaps our best chance to make your extension work is to fix the extension itself trying to imitate the bookkeeping that the `import` tag does to support `--incremental` behaviour.

**Comment by jonsterling** [comment]
i like the idea of fixing my extension to do the proper bookkeeping…

**Comment by jonsterling** [comment]
But i would say this is not a huge priority for me right now.
